mediathread/discussions/views.py

```diff
@@ -129,20 +132,23 @@ def discussion_create(request):
 
 class DiscussionDeleteView(LoggedInFacultyMixin, View):
     def post(self, request, discussion_id):
-        root_comment = get_object_or_404(ThreadedComment, pk=discussion_id)
+        discussion = get_object_or_404(DiscussionIndex, pk=discussion_id)
```

Comment:
I changed this because I thought it didn't make sense how we were looking up a ThreadedComment by its discussion ID. The tests were failing here, and this resolved it. But, I this will also be resolved by changing the comment PROTECT on_delete attribute to CASCADE, like you mentioned above. So, I'll revert this to how it was originally working.

Reply:
DiscussionIndex is this other "thing" that was setup to support the course feed...which we don't really use anymore. I haven't looked at it in years, but it was basically constructed as a denormalized data view. It's unrelated to ThreadedComments. The `discussion_id` here references the parent-level ThreadedComment.